flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -116,15 +116,28 @@ object CodeGenUtils {
 
   // ----------------------------------------------------------------------------------------
 
-  private val nameCounter = new AtomicInteger
+  private val nameCounter = new AtomicLong
+
+  private def newNameSuffixNumber(): Long = {
+    val newNameCounter = nameCounter.getAndIncrement
+    if (newNameCounter < 0) {
+      // should we throw a Exception here in case duplicate suffix number ?
+      synchronized {
+        nameCounter.set(0)
+      }
+      nameCounter.getAndIncrement
+    } else {
+      newNameCounter
+    }
+  }
 
   def newName(name: String): String = {
-    s"$name$$${nameCounter.getAndIncrement}"
+    s"$name$$${newNameSuffixNumber()}"
```

Comment:
In online service performance is important, extra function call will cause performance degradation. Check less than zero is  not necessary here. On the other side, newName() should return a unique class name here.

Reply:
Sorry just saw the comment,i agree with you,but check less than zero may be very lightweith,
and we should generate a unique class name in theoretically,but the generate classes exceed Long.MAX_VALUE may be also a problem

Reply:
I don't agree with you for some reasons below:
1) extra function call is slow several times than atomic increment, not lightweight;
2) generate duplicated class name will also throw exception when compile;
3) For Long.MAX_VALUE, if service serves 500 sqls per second and 3 expression reducers per sql, it will take 197688872 years until overflow;   // Long.MAX_VALUE/((long)12 * 30 * 24 * 3600 * 500 * 3)

Reply:
yes I think so